app/controllers/batch_update_tools_controller.rb

```diff
@@ -9,26 +10,58 @@ def show
                 else
                   []
                 end
+
+    classes_file = File.open("app/assets/data/classes.json")
+    classes_data = classes_file.read
+    @classes = JSON.parse(classes_data)
```

Comment:
This can be shortened to a one-liner to keep some of the noise down:

```suggestion
    @classes = JSON.parse(File.read("app/assets/data/classes.json"))
```

I might make this a method since you do it in both controller actions.

Reply:
fair enough